src/ln/msgs.rs

```diff
@@ -1682,6 +1702,594 @@ impl MsgDecodable for ErrorMessage {
 	}
 }
 
+impl_writeable!(AcceptChannel, {
+	temporary_channel_id,
+	dust_limit_satoshis,
+	max_htlc_value_in_flight_msat,
+	channel_reserve_satoshis,
+	htlc_minimum_msat,
+	minimum_depth,
+	to_self_delay,
+	max_accepted_htlcs,
+	funding_pubkey,
+	revocation_basepoint,
+	payment_basepoint,
+	delayed_payment_basepoint,
+	htlc_basepoint,
+	first_per_commitment_point,
+	shutdown_scriptpubkey
+});
+
+impl_writeable!(AnnouncementSignatures, {
+	channel_id,
+	short_channel_id,
+	node_signature,
+	bitcoin_signature
+});
+
+impl<W: ::std::io::Write> Writeable<W> for ChannelReestablish {
+	fn write(&self, w: &mut Writer<W>) -> Result<(), DecodeError> {
+		self.channel_id.write(w)?;
+		self.next_local_commitment_number.write(w)?;
+		self.next_remote_commitment_number.write(w)?;
+		if let Some(ref data_loss_protect) = self.data_loss_protect {
+			data_loss_protect.your_last_per_commitment_secret.write(w)?;
+			data_loss_protect.my_current_per_commitment_point.write(w)?;
+		}
+		Ok(())
+	}
+}
+
+impl<R: ::std::io::Read> Readable<R> for ChannelReestablish{
+	fn read(r: &mut Reader<R>) -> Result<Self, DecodeError> {
+		Ok(Self {
+			channel_id: Readable::read(r)?,
+			next_local_commitment_number: Readable::read(r)?,
+			next_remote_commitment_number: Readable::read(r)?,
+			data_loss_protect: {
+				let mut buf = [0; 32+33];
+				match r.read_exact(&mut buf) {
+					Ok(()) => Some(DataLossProtect {
+						your_last_per_commitment_secret: {
```

Comment:
Because you aren't allows to only fill in part of data_loss_protect you should be able to simplify this a bit by swapping buf for  your_last_per_commitment_secret directly, then just calling Readable::read(r)? for my_current_per_commitment_point.